router/customdestinationmanager/customdestinationmanager_test.go

```diff
@@ -144,3 +145,62 @@ func TestSendDataWithStreamDestination(t *testing.T) {
 	mockProducer.EXPECT().Produce(event, someDestination.Config).Times(1)
 	customManager.SendData(event, someDestination.ID)
 }
+
+func TestKVManagerHSETInvocation(t *testing.T) {
+	initCustomerManager()
+	customManager := New("REDIS", Opts{}).(*CustomManagerT)
+	someDestination := backendconfig.DestinationT{
+		ID: "someDestinationID1",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			Name: "REDIS",
+		},
+	}
+	err := customManager.onNewDestination(someDestination)
+	assert.Nil(t, err)
+
+	event := json.RawMessage(`{
+		"message": {
+			"key": "someKey",
+			"value" : "someValue",
+			"hash": "someHash"
+		}
+	  }
+	`)
+	ctrl := gomock.NewController(t)
+	mockKVStoreManager := mock_kvstoremanager.NewMockKVStoreManager(ctrl)
+	mockKVStoreManager.EXPECT().HSet("someHash", "someKey", "someValue").Times(1)
+	mockKVStoreManager.EXPECT().StatusCode(nil).Times(1)
+	customManager.send(event, mockKVStoreManager, someDestination.Config)
+}
+
+func TestKVManagerHMSETInvocation(t *testing.T) {
+	initCustomerManager()
+	customManager := New("REDIS", Opts{}).(*CustomManagerT)
+	someDestination := backendconfig.DestinationT{
+		ID: "someDestinationID1",
+		DestinationDefinition: backendconfig.DestinationDefinitionT{
+			Name: "REDIS",
+		},
+	}
+	err := customManager.onNewDestination(someDestination)
+	assert.Nil(t, err)
+
+	event := json.RawMessage(`{
+		"message": {
+			"key": "someKey",
+			"fields" : {
+				"field1": "value1",
+				"field2": "value2"
+			}
+		}
+	  }
+	`)
+	ctrl := gomock.NewController(t)
+	mockKVStoreManager := mock_kvstoremanager.NewMockKVStoreManager(ctrl)
+	mockKVStoreManager.EXPECT().HMSet("someKey", map[string]interface{}{
+		"field1": "value1",
+		"field2": "value2",
+	}).Times(1)
+	mockKVStoreManager.EXPECT().StatusCode(nil).Times(1)
+	customManager.send(event, mockKVStoreManager, someDestination.Config)
+}
```

Comment:
[Optional] Maybe we can remove duplicate code and compact these tests?

```suggestion
func TestKVManagerInvocation(t *testing.T) {
	initCustomerManager()

	customManager := New("REDIS", Opts{}).(*CustomManagerT)
	someDestination := backendconfig.DestinationT{
		ID: "someDestinationID1",
		DestinationDefinition: backendconfig.DestinationDefinitionT{
			Name: "REDIS",
		},
	}
	err := customManager.onNewDestination(someDestination)
	assert.Nil(t, err)

	ctrl := gomock.NewController(t)
	defer ctrl.Finish()

	t.Run("HSET", func(t *testing.T) {
		event := json.RawMessage(`{
			"message": {
				"key": "someKey",
				"value" : "someValue",
				"hash": "someHash"
			}
		  }
		`)

		mockKVStoreManager := mock_kvstoremanager.NewMockKVStoreManager(ctrl)
		mockKVStoreManager.EXPECT().HSet("someHash", "someKey", "someValue").Times(1)
		mockKVStoreManager.EXPECT().StatusCode(nil).Times(1)
		customManager.send(event, mockKVStoreManager, someDestination.Config)
	})
	t.Run("HMSET", func(t *testing.T) {
		event := json.RawMessage(`{
			"message": {
				"key": "someKey",
				"fields" : {
					"field1": "value1",
					"field2": "value2"
				}
			}
		  }
		`)

		mockKVStoreManager := mock_kvstoremanager.NewMockKVStoreManager(ctrl)
		mockKVStoreManager.EXPECT().HMSet("someKey", map[string]interface{}{
			"field1": "value1",
			"field2": "value2",
		}).Times(1)
		mockKVStoreManager.EXPECT().StatusCode(nil).Times(1)
		customManager.send(event, mockKVStoreManager, someDestination.Config)
	})
}
```

Reply:
addressed